### DISCUSSION TITLE
Station precipitation value rounding issue?

### DISCUSSION THREAD

**Discussion by maplegilbs** [weather-gov/api#494] [Q&A]
We had a very rainy halloween here in Vermont, and when I checked on my program fetching station observation data from around VT I realized values were off from the 3-day history from the website.

I can only comment on stations in VT, but it appears the observations endpoint for individual stations is not reporting hourly rainfall correctly. 

This is referring to the stationId/observations endpoint and specifically the value at 
features.'observation number'.properties.precipitationLastHour.value though I would imagine it is affecting all precipitation values.

I suspect the error is a rounding issue combined with the fact rainfall is reported in meters.  If I had to guess, these precipitationLastHour values are being rounded at the thousandths place and then reported at the hundreths.  I.e .005 M of rainfall will be rounded to .01 and reported... as such any values of less than .20 inches (~.005M) will be rounded down and not be reported.  My reasoning for this is comparing the datasets below.  Note the only times the API data reported rainfall was 14:54 & 13:54 (UTC) which corresponds with the data from the 3-day history at 10:54 & 09:54 (EST) both of which were the only totals from these datasets >= .20 inches (.005M).

Hopefully this is helpful in resolving the issue.  Let me know if I can provide any more assistance.  

For example at the KBTV station
https://api.weather.gov/stations/KBTV/observations

Here is a sorted print of the observations for precipitationLastHour from 10/31/2021
![image](https://user-images.githubusercontent.com/88692608/139604992-b1aa7a7c-e85e-4c86-a5cb-288cb267c0f4.png)

And here is from the 3-day history of the same site: https://w1.weather.gov/data/obhistory/KBTV.html
![image](https://user-images.githubusercontent.com/88692608/139605019-6de61c54-2c09-4d33-af5c-00a1ef1d1f51.png)

Anybody have thoughts on how to resolve - @StephenClouse - sorry to bug, sure you are plenty busy.

Greatly appreciated,
Scott

**Comment by jsvnsk** [comment]
I agree the observation history hourly precipitation data seems to be off.

I consistently see only the following 4 values for **precipitationLastHour**, **precipitationLast3Hours**, and **precipitationLast6Hours**, regardless of station:

1. empty / null
2. 0
3. 0.01 (when converted to inches and rounded to two decimal places appears as 0.39")
4. 0.02 (when converted to inches and rounded to two decimal places appears as 0.79")

That is, no values other than empty/null, 0, 0.01, or 0.02 seem to appear in those three fields, regardless of station.

**Comment by maplegilbs** [comment]
Just FYI to anybody looking for this fix, I have been told the ticket is slated for Feb/Mar 2022 to be addressed

**Reply by jah1007** [reply]
The total precip parameters I have confirmed are an upstream data issue, at least the 6 and 24 hour should be more reliably filled in later this year. One note on 24 hour max temp though, it will only ever be filled in on the observation that covers local midnight, it is not an hourly reported data parameter.

**Reply by maplegilbs** [reply]
Any word on the resolution of this?  Still seeing the issue.

**Reply by jah1007** [reply]
This appears to be tagged for an API release this summer.
